### ISSUE TITLE
Make Design somewhat responsive

### ISSUE THREAD

**Issue by tolik518** [tolik518/blog.returnnull.de#19]
The design looks currently pretty awful on mobile. 

This should be fixed as studies show that websites usually gain more traffic through mobile than desktop. 

The design should be made somewhat acceptable for mobile.

**Comment by pavelee** [comment]
To make it responsive I would recommend:

- We should define new a template with a layout. All other projectors could just use it to fill content into it.
   - Layout templates takes care to be mobile friendly
   - We could move there menu rendering etc.
- to use flex system -> https://developer.mozilla.org/en-US/docs/Web/CSS/flex
- get rid of using margins as way to structure grid system, we should use gap -> https://developer.mozilla.org/en-US/docs/Web/CSS/gap
- use media screen queries -> https://developer.mozilla.org/en-US/docs/Web/CSS/@media#syntax 


There is also other way 🫣, integrate https://twig.symfony.com, this is much more refactor but it's ready to go template engine 🥳, but it's not solving css styles problem 😅

**Comment by tolik518** [comment]
@pavelee Thanks a lot for your comment!

When I initially created the CSS for the blog, my experience with web development and CSS was very limited. As a result, the current CSS structure is somewhat chaotic and inflexible 😅

Relying heavily on margins and paddings has indeed made the CSS very static and not fun to mess with. Implementing gaps and flex boxes would be a shift in the right direction. 

I'm not much of a fan of twig though, since the projector/page-pattern that's used in the project pretty much gets rid of logic-handling directly in the template. This is a personal preference - but I don't like having a templating engine 😅

**Comment by pavelee** [comment]
I could try to fix the css, just to make layout mobile friendly. 

Then we cloud address the template engine issue. Maybe write our own 🤓

**Comment by tolik518** [comment]
Yeah, an own templating engine would make sense in the margins of the project - it could be a good learning experience :) 

If you'd like to fix the CSS, you are very welcome, go ahead! I'll assign you to the issue.

###### A bit offtopic, but I just found my old todo-list for the project, I'm gonna open some issues here

**Comment by pavelee** [comment]
Sorry for no actions, i have a lot work right now (end of the year 😅)

I will back to the topic in the next year 🫡

**Comment by tolik518** [comment]
@pavelee don't worry, we all have stuff to do :)
